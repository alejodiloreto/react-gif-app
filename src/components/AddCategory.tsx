import { useState } from 'react';

interface Props {
  setCategories: React.Dispatch<React.SetStateAction<string[]>>;
}

export const AddCategory: React.FC<Props> = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState<string>('');

  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    setCategories((categories) => [inputValue, ...categories]);
    setInputValue('');
  };

  return (
    <form onSubmit={handleSubmit} aria-label='form'>
      <input type="text" value={inputValue} onChange={handleInputChange} name='input-text' />
    </form>
  );
};
