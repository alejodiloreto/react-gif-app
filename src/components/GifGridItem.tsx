
export interface Props {
  title: string;
  url: string;
}

export const GifGridItem: React.FC<Props> = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <h5>{title}</h5>
      <img src={url} alt={title} />
    </div>
  );
};
