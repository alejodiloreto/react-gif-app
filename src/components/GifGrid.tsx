import { useFetchGifs } from '../hooks/useFetchGifs';
import { Gif } from '../interfaces/gif';
import { GifGridItem } from './GifGridItem';

interface Props {
  category: string;
}

export const GifGrid: React.FC<Props> = ({ category }) => {

  const { data, loading } = useFetchGifs(category);

  return (
    <article>
      <h3>{category}</h3>

      {loading && <p className="animate__animated animate__flash">Cargando...</p>}

      <div className="card-grid">
        {data.map((item: Gif) => (
          <GifGridItem key={item.id} {...item} />
        ))}
      </div>
    </article>
  );
};
