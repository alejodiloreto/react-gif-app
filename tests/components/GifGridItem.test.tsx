import { render, screen } from "@testing-library/react"
import { GifGridItem } from '../../src/components/GifGridItem';

describe('<GifGridItem />', () => {

  const title = 'titulo';
  const url = 'url';

  it('should render <GifGridItem />', () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  })

  it('should show image with an URL and Alt', () => {
    render(<GifGridItem title={title} url={url} />);
    const { src, alt } = screen.getByRole('img') as HTMLImageElement;
    expect(src).toMatch(new RegExp(url + '$'));
    expect(alt).toBe(title);
  })

  it('should show the component title', () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  })

})