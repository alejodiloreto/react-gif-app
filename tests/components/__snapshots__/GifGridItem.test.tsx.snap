// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> should render <GifGridItem /> 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn"
  >
    <h5>
      titulo
    </h5>
    <img
      alt="titulo"
      src="url"
    />
  </div>
</div>
`;
